### PR TITLE
fix readme code samples to show working examples

### PR DESCRIPTION
**Pull request by pavlovic-ivan**
Simply fixing the code example so that they can be simply copy-pasted after fixes that Adam is working on

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -220,32 +220,46 @@ Run the example in one line:
...
-      Float8Vector vector = (Float8Vector) root.getVector(0);
-      for (int i = 0; i < root.getRowCount(); i += 1) {
-        logger.info("value {}={}", i, vector.getValueAsDouble(i));
-      }
+    private static final Logger logger = LoggerFactory.getLogger(ExampleMain.class);
+
+    public static void main(String[] args) throws Exception {
+        try (SessionContext context = SessionContexts.create(); BufferAllocator allocator = new RootAllocator()) {
+            CompletableFuture<DataFrame> dataFrame = context.sql("select 1.5 + sqrt(2.0)");
+            dataFrame.get().collect(allocator).thenAccept(ExampleMain::onReaderResult);
```
Comment (adamreeve) [inline README.md]: I think we need a `.get()` on the `thenAccept` result to make sure the `onReaderResult` method has completed. Might also be tidier to move the get call for the data frame creation:
```suggestion
            DataFrame dataFrame = context.sql("select 1.5 + sqrt(2.0)").get();
            dataFrame.collect(allocator).thenAccept(ExampleMain::onReaderResult).get();
```

File: README.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ import org.apache.arrow.datafusion.SessionContexts;
 public class DataFusionDemo {
 
     public static void main(String[] args) throws Exception {
-        try (ExecutionContext executionContext = ExecutionContexts.create()) {
+        try (SessionContext executionContext = SessionContexts.create()) {
```
Comment (adamreeve) [inline README.md]: Should probably rename `executionContext` too

File: README.md (hunk 1 above)
Comment (pavlovic-ivan) [reply README.md]: sounds good

File: README.md (hunk 2 above)
Comment (pavlovic-ivan) [reply README.md]: will do. i will change it in both examples, to keep it consistent

**Review by adamreeve — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (pavlovic-ivan) [reply README.md]: @adamreeve changed accordingly. please resolve conversation if you agree with the changes

File: README.md (hunk 2 above)
Comment (pavlovic-ivan) [reply README.md]: @adamreeve changed accordingly. please resolve conversation if you agree with the changes

**Review by adamreeve — APPROVED** [review]

**Review by jgiannuzzi — APPROVED** [review]

**Pull request merged**